core/src/main/scala/org/apache/spark/rpc/netty/Dispatcher.scala

```diff
@@ -144,24 +144,21 @@ private[netty] class Dispatcher(nettyEnv: NettyRpcEnv) extends Logging {
       endpointName: String,
       message: InboxMessage,
       callbackIfStopped: (Exception) => Unit): Unit = {
-    val shouldCallOnStop = synchronized {
+    val error: Option[Exception] = synchronized {
       val data = endpoints.get(endpointName)
-      if (stopped || data == null) {
-        true
+      if (stopped) {
+        Some(new RpcEnvStoppedException())
+      } else if (data == null) {
+        Some(new SparkException(s"Could not find $endpointName."))
       } else {
         data.inbox.post(message)
         receivers.offer(data)
-        false
+        None
       }
     }
-    if (shouldCallOnStop) {
+    if (error.isDefined) {
```

Comment:
This could be `error.foreach(callbackIfStopped)`. You don't need the `Option` type above I think, but up to your judgment about readability.